Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionAssetPropertyDrawer.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+using UnityEditor;
+using UnityEditor.SearchService;
+using UnityEngine.Search;
+using UnityEngine.UIElements;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    // [CustomPropertyDrawer(typeof(InputActionAsset))]
+    [CustomPropertyDrawer(typeof(InputActionAssetProperty))]
```

Comment:
It would be highly desirable to have this as [CustomPropertyDrawer(typeof(InputActionAsset))] instead. Then any third-party or other package code would just work, while this approach may enable assigning project-wide or asset it has the drawback of requiring anyone interested in referencing project wide to update/adapt their code. 

I did a quick test with CustomPropertyDrawer(typeof(InputActionAsset))] and that enables a drawer but leaves you with the problem of figuring out what is really selected to be able to set the value of the first high-level drop-down, which leads me to my next question:
How do we currently reference the project-wide asset? Is it available as a static instance somewhere? If it is, and I interpret InputActionAsset as basically being a reference (project wide is serialised independently by the settings?), why can't we leave PlayerInput and PlayerINputEditor unmodified and instead provide a new property on the InputActionAsset (evaluated by the drawer) which is bool IsProjectWide().
That getter could in turn base the result, ideally on reference comparison with the project wide instance, or on detecting a tag from json parsing (InputActionAsset.FromJson) not serialised for any other asset. What I am after is that any solution not introducing a new type here would be desirable in my perspective.

Reply:
My initial idea was to do a custom drawer for `InputActionAsset` but after not being able to make it work properly I decided with a different strategy, and found that this pattern would do what we wanted.
That idea sounds interesting and I'll try it out, thanks. 👍

Reply:
Good proposal and seems reasonable if this works.

Reply:
Addressed in newer commits. Thanks @ekcoh for the suggestion